study-group/Toast-Implementation-Studies.md

```diff
@@ -90,9 +90,132 @@ The `restoreFocusOnClear` attribute restores page's focus once the toast is clea
 
-*TODO: Fill in study for this library*
+
+### Design & Sample Code
+Before using ngx-toastr, setup is required. First, add the CSS:
+
+```js
+// regular style toast
+@import '~ngx-toastr/toastr';
+
+// bootstrap style toast
+@import '~ngx-toastr/toastr-bs4-alert';
+```
+
+Next, add ToastrModule and BrowserAnimationsModule to app NgModule:
+
+```js
+import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
+import { ToastrModule } from 'ngx-toastr';
+
+@NgModule({
+  imports: [
+    CommonModule,
+    BrowserAnimationsModule, // required animations module
+    ToastrModule.forRoot() // ToastrModule added
+  ],
+  ...
+})
+...
+```
+
+Once the modules are set up, this code will display a toast:
+
+```js
+import { ToastrService } from 'ngx-toastr';
+
+@Component({...})
+export class SomeComponent {
+  constructor(private toastr: ToastrService) {}
+
+  showSuccess() {
+    this.toastr.success('Hello title!', 'Hello message!', { timeOut: 3000 });
+  }
+}
+```
+
+To show the toast with an instance of ToastrService, use `toastr.success/error/warning/info/show()`, depending on the context and intent.
+The third parameter configures the toast with a variety of specialized options, for example:
+
+```js
+this.toastrService.error('Uh Oh...', '<b>Something is Wrong<b>', {
+  timeOut: 20000,
+  tapToDismiss: false,
+  enableHtml: true,
+  closeButton: true,
+  progressBar: true
+});
+```
+
+All these options can be configured globally like so:
+
+```js
+// root app NgModule
+imports: [
+  ToastrModule.forRoot({
+    timeOut: 10000,
+    positionClass: 'toast-bottom-right',
+    preventDuplicates: true,
+  }),
+],
+```
+
+These global options will be overriden by the individual toast configurations.
+For finer control, ngx-toaster allows both custom toast components and display in container, used like this:
+
+```js
+// root app NgModule
+import { ToastrModule, ToastContainerModule } from 'ngx-toastr';
+
+@NgModule({
+  imports: [
+    ToastrModule.forRoot({
+      toastComponent: YourToastComponent // added custom toast!
+    })
+  ],
+  entryComponents: [YourToastComponent], // add!
+  bootstrap: [App],
+  declarations: [App, YourToastComponent] // add!
+})
+...
+// App
+@Component({
+  template: `
+  <h1><a (click)="onClick()">Click</a></h1>
+  <div toastContainer></div>`
+})
+export class AppComponent implements OnInit {
+  @ViewChild(ToastContainerDirective) toastContainer: ToastContainerDirective;
+
+  constructor(private toastrService: ToastrService) {}
+  ngOnInit() {
+    this.toastrService.overlayContainer = this.toastContainer;
+  }
+  onClick() {
+    this.toastrService.success('Custom in a container!');
+  }
+}
+```
+
+### Notable Features and Details
+
+#### Animation
+Uses Angular [Web Animations API](https://angular.io/guide/animations) by default.
+
+#### Captures User Hover
+User hover pauses the toast timeout, and removing hover replaces timeout with an optional different timeout.
```

Comment:
Out of curiosity, since hover is basically non-existent on mobile, is there a way to pause the timeout on mobile?

Reply:
That's a good point and I'll make a note of that in the takeaways section. I tested on my phone and I couldn't get it to persist. There's no simple way, I think the closest solution for mobile (with this library) would be override the `onTap` action so the first tap keeps the toast up and the second tap dismisses it, but it's a bit of a hack and not the exact behavior we're looking for.